packages/jest-haste-map/src/crawlers/node.js

```diff
@@ -0,0 +1,121 @@
+ /**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+'use strict';
+
+const H = require('../constants');
+
+const fs = require('fs');
+const path = require('path');
+const spawn = require('child_process').spawn;
+
+function find(roots, extensions, ignore, callback) {
+  const result = [];
+  let activeCalls = 0;
+
+  function search(directory) {
+    activeCalls++;
+    fs.readdir(directory, (err, names) => {
+      activeCalls--;
+
+      names.forEach(file => {
+        file = path.join(directory, file);
+        if (ignore(file)) {
+          return;
+        }
+        activeCalls++;
+
+        fs.lstat(file, (err, stat) => {
+          activeCalls--;
+
+          if (!err && stat && !stat.isSymbolicLink()) {
+            if (stat.isDirectory()) {
+              search(file);
+            } else {
+              const ext = path.extname(file).substr(1);
+              if (extensions.indexOf(ext) !== -1) {
+                result.push([file, stat.mtime.getTime()]);
+              }
+            }
+          }
+          if (activeCalls === 0) {
+            callback(result);
+          }
+        });
+      });
+
+      if (activeCalls === 0) {
+        callback(result);
+      }
+    });
+  }
+
+  roots.forEach(search);
+}
+
+function findNative(roots, extensions, ignore, callback) {
+  const args = [].concat(roots);
+  args.push('-type', 'f');
+  extensions.forEach((ext, index) => {
```

Comment:
wait, what's the difference? It seems to work fine on www for example. This is what it was like in node-haste1.